src/App.tsx

```diff
@@ -17,18 +46,36 @@ export const App: React.FC = () => {
             <h1 className="title">Todos:</h1>
 
             <div className="block">
-              <TodoFilter />
+              <TodoFilter
+                onUploadedTodos={todosFromServer}
+                onCurrentTodos={setFilteredTodos}
+              />
             </div>
 
             <div className="block">
-              <Loader />
-              <TodoList />
+              {hasError
+                && <span>Error 404. Please try later.</span>}
+              {isLoaded
+                ? !hasError && (
+                  <TodoList
+                    todos={filteredTodos}
+                    onSelectTodo={handleTodo}
+                    onClosedTodo={selectedTodo}
+                  />
+                )
+                : <Loader />}
```

Comment:
Instead of this weird condition mixing split this into two separate expressions: one for loading, one for TodoList.